src/Cafe/OS/libs/nn_fp/nn_fp.cpp

```diff
@@ -607,6 +607,14 @@ namespace nn
 			return resultBuf != 0 ? 1 : 0;
 		}
 
+		nnResult UpdateCommentAsync(char16_t* newComment, void* funcPtr, void* customParam)
+		{
+			FP_API_BASE();
+			auto ipcCtx = std::make_unique<FPIpcContext>(iosu::fpd::FPD_REQUEST_ID::UpdateCommentAsync);
+			ipcCtx->AddInput(newComment, 36);
```

Comment:
`newComment` is a null-terminated big-endian wide string and our current convention for those is to define them as `uint16be*`
But also I took a quick look into nn_fp.rpl and it behaves differently here. The length passed to `AddInput` is dynamically calculated. I think this is how its implemented:
```
uint32 commentLen = CafeStringHelpers::Length(newComment, 17);
if(commentLen>=17)
    return FPResult_InvalidIPCParam;
ipcCtx->AddInput(newComment, sizeof(uint16be) * (commentLen + 1));    
```
So it seems to work similar to how `UpdateGameModeWithUnusedParam` passes it's string.

Reply:
I checked nn_fp.rpl and yeah, that implementation is accurate, though commentLen is added by 2 instead of 1